File: template-engine/src/main/java/com/github/twitch4j/codegen/engine/pebble/CodeGenPebbleExtension.java
package com.github.twitch4j.codegen.engine.pebble;

import com.github.twitch4j.codegen.engine.pebble.filter.LinePrefixFilter;
import com.github.twitch4j.codegen.engine.pebble.filter.PadRightFilter;
import com.github.twitch4j.codegen.engine.pebble.filter.WrapInFilter;
import com.github.twitch4j.codegen.engine.pebble.function.GetOrDefaultFunction;
import com.github.twitch4j.codegen.engine.pebble.function.NewLineFunction;
import com.github.twitch4j.codegen.engine.pebble.operator.StartsWithOperator;
import io.pebbletemplates.pebble.attributes.AttributeResolver;
import io.pebbletemplates.pebble.extension.Extension;
import io.pebbletemplates.pebble.extension.Filter;
import io.pebbletemplates.pebble.extension.Function;
import io.pebbletemplates.pebble.extension.NodeVisitorFactory;
import io.pebbletemplates.pebble.extension.Test;
import io.pebbletemplates.pebble.operator.BinaryOperator;
import io.pebbletemplates.pebble.operator.UnaryOperator;
import io.pebbletemplates.pebble.tokenParser.TokenParser;

import java.util.List;
import java.util.Map;

public class CodeGenPebbleExtension implements Extension {

    @Override
    public Map<String, Filter> getFilters() {
        return Map.of(
                "padright", new PadRightFilter(),
                "wrapin", new WrapInFilter(),
                "lineprefix", new LinePrefixFilter()
        );
    }

    @Override
    public Map<String, Test> getTests() {
        return null;
    }

    @Override
    public Map<String, Function> getFunctions() {
        return Map.of(
                "newline", new NewLineFunction(),
                "getOrDefault", new GetOrDefaultFunction()
        );
    }

    @Override
    public List<TokenParser> getTokenParsers() {
        return null;
    }

    @Override
    public List<BinaryOperator> getBinaryOperators() {
        return List.of(new StartsWithOperator());
    }

    @Override
    public List<UnaryOperator> getUnaryOperators() {
        return null;
    }

    @Override
    public Map<String, Object> getGlobalVariables() {
        return null;
    }

    @Override
    public List<NodeVisitorFactory> getNodeVisitors() {
        return null;
    }

    @Override
    public List<AttributeResolver> getAttributeResolver() {
        return null;
    }
}

File: template-engine/src/main/java/io/github/primelib/primecodegen/templateengine/pebble/CodeGenPebbleExtension.kt
package io.github.primelib.primecodegen.templateengine.pebble;

import io.github.primelib.primecodegen.templateengine.pebble.filter.LinePrefixFilter;
import io.github.primelib.primecodegen.templateengine.pebble.filter.PadRightFilter;
import io.github.primelib.primecodegen.templateengine.pebble.filter.WrapInFilter;
import io.github.primelib.primecodegen.templateengine.pebble.function.GetOrDefaultFunction;
import io.github.primelib.primecodegen.templateengine.pebble.function.JavadocDescription
import io.github.primelib.primecodegen.templateengine.pebble.function.JavadocParamDescription
import io.github.primelib.primecodegen.templateengine.pebble.function.NewLineFunction;
import io.github.primelib.primecodegen.templateengine.pebble.operator.StartsWithOperator;
import io.pebbletemplates.pebble.attributes.AttributeResolver;
import io.pebbletemplates.pebble.extension.Extension;
import io.pebbletemplates.pebble.extension.Filter;
import io.pebbletemplates.pebble.extension.Function;
import io.pebbletemplates.pebble.extension.NodeVisitorFactory;
import io.pebbletemplates.pebble.extension.Test;
import io.pebbletemplates.pebble.operator.BinaryOperator;
import io.pebbletemplates.pebble.operator.UnaryOperator;
import io.pebbletemplates.pebble.tokenParser.TokenParser;

class CodeGenPebbleExtension : Extension {
    override fun getFilters(): Map<String, Filter> {
        return mapOf(
            "padright" to PadRightFilter(),
            "wrapin" to WrapInFilter(),
            "lineprefix" to LinePrefixFilter()
        )
    }

    override fun getTests(): Map<String, Test>? {
        return null
    }

    override fun getFunctions(): Map<String, Function> {
        return mapOf(
            "newline" to NewLineFunction(),
            "getOrDefault" to GetOrDefaultFunction(),
            "javadocParamDescription" to JavadocParamDescription(),
            "javadocDescription" to JavadocDescription(),
        )
    }

    override fun getTokenParsers(): List<TokenParser>? {
        return null
    }

    override fun getBinaryOperators(): List<BinaryOperator> {
        return listOf(StartsWithOperator())
    }

    override fun getUnaryOperators(): List<UnaryOperator>? {
        return null
    }

    override fun getGlobalVariables(): Map<String, Any>? {
        return null
    }

    override fun getNodeVisitors(): List<NodeVisitorFactory>? {
        return null
    }

    override fun getAttributeResolver(): List<AttributeResolver>? {
        return null
    }
}
